platform/storage/zephyr/nvs/src/mender-storage.c

```diff
@@ -39,6 +39,7 @@
 #define MENDER_STORAGE_NVS_DEPLOYMENT_DATA 3
 #define MENDER_STORAGE_NVS_DEVICE_CONFIG   4
 #define MENDER_STORAGE_NVS_PROVIDES        5
+#define MENDER_STORAGE_NVS_NEW_PROVIDES    6
```

Comment:
The "new provides" is really part of the Deployment Data -> info about the ongoing deployment with an uncommitted Artifact. Can't we extend the existing key for that?

Reply:
I could probably do that; I can add the parsed provides data as a string under a key like "new_provides" in `deployment-data.json`

Reply:
That would make more sense to me, and would align with the C++ client. For example if we consolidate we can clear both in the clean state, or check only one place to assume there is or there is not a deployment in progress.

Reply:
Yeah, I didn't think about doing it that way, but that makes a lot of sense.